### DISCUSSION TITLE
Error : php artisan snapshot: load db_name

### DISCUSSION THREAD

**Discussion by jnkodo** [spatie/laravel-db-snapshots#133] [Q&A]
Symfony\Component\ErrorHandler\Error\FatalError

  Allowed memory size of 536870912 bytes exhausted (tried to allocate 186770585 bytes)
![image](https://user-images.githubusercontent.com/116576180/201483629-9adda0fb-9b96-42f0-9a37-3cd08c6996da.png)

**Comment by TheDaveKent** [comment]
Probably, you figured this out but you want to use the stream option:

`php artisan snapshot:load danapay-api-dump --stream`

By default, snapshot:load will load the entire snapshot into memory which may cause problems when using large files. To avoid this, you can pass the --stream option to stream the snapshot to the database one statement at a time.

**Comment by sebastiaanluca** [comment]
I had an issue where INSERT statements (restoring db) where failing without a real reason (export syntax was ok). Figured out MySQL couldn't load the query (single line) in memory and just failed halfway through.

This fixed it (in `config/database.php` on the MySQL connection):

```php
            'dump' => [
                // Prevent long INSERT statements when snapshotting database (25MB)
                'addExtraOption' => '--max-allowed-packet=25000000 --net_buffer_length=25000000',
            ],
```

Export again after this. It sets the maximum buffer to 25MB so single-line INSERT and other statements don't exceed that limit.
